frontend/src/components/Pages/FrontPage/MissionOverview/StopDialogs.tsx

```diff
@@ -50,11 +52,47 @@ export enum MissionStatusRequest {
     Resume,
 }
 
-export const StopMissionDialog = ({ missionName, robotId }: MissionProps): JSX.Element => {
+export const StopMissionDialog = ({ missionName, robotId, missionTaskType }: MissionProps): JSX.Element => {
     const { TranslateText } = useLanguageContext()
     const [isStopMissionDialogOpen, setIsStopMissionDialogOpen] = useState<boolean>(false)
     const { updateRobotMissionState } = useMissionControlContext()
 
+    const dialogContent = (missionTaskType: TaskType) => {
```

Comment:
Suggest moving this function out from the function "StopMissionDialog" to avoid long functions. If moved it should also be renamed to show that it's only used for stopMissionDialog

Reply:
I don't think it should be moved out. It is used to select the text in the stop dialogs present in this file, having it outside doesn't make sense to me

Reply:
But why should it be inside StopMissionDialog if it doesn't need to be? We can still have it be a function in the same file, but nesting it in the other function seems unneccessary to me